lib/node_modules/@stdlib/assert/deep-has-own-property/lib/factory.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isArray = require( '@stdlib/assert/is-array' );
+var isObjectLike = require( '@stdlib/assert/is-object-like' );
+var copy = require( '@stdlib/utils/copy' );
+var validate = require( './validate.js' );
+var defaults = require( './defaults.json' );
+var has = require( './d_has_own.js' );
+
+
+// MAIN //
+
+/**
+* Returns a function which checks whether an object has a nested key path.
+*
+* @param {(string|Array)} path - key path
+* @param {Options} [options] - function options
+* @param {string} [options.sep='.'] - key path separator
+* @throws {TypeError} first argument must be a string primitive or key array
+* @throws {TypeError} options argument must be an object
+* @throws {TypeError} must provide valid options
+* @returns {Function} function which checks whether an object has a nested key path
+*
+* @example
+* var has = factory( 'a/b/c', {
+*     'sep': '/'
+* });
+*/
+function factory( path, options ) {
+	var isStr;
+	var props;
+	var opts;
+	var err;
+	isStr = isString( path );
+	if ( !isStr && !isArray( path ) ) {
+		throw new TypeError( 'invalid input argument. Key path must be a string primitive or a key array. Value: `' + path + '`.' );
+	}
+	opts = copy( defaults );
+	if ( arguments.length > 1 ) {
+		err = validate( opts, options );
+		if ( err ) {
+			throw err;
+		}
+	}
+	if ( isStr ) {
+		props = path.split( opts.sep );
+	} else {
+		props = path;
+	}
+	return deepHas;
+
+	/**
+	* Returns a boolean indicating whether an object has a nested key path.
+	*
+	* @param {ObjectLike} obj - input object
```

Comment:
Add `@private` JSDoc annotation (which is missing in `deep-get` and `deep-set`).